packages/eslint-plugin-query/src/rules/stable-query-client/stable-query-client.rule.ts

```diff
@@ -0,0 +1,90 @@
+import { AST_NODE_TYPES } from '@typescript-eslint/utils'
+import { ASTUtils } from '../../utils/ast-utils'
+import { createRule } from '../../utils/create-rule'
+import type { TSESLint } from '@typescript-eslint/utils'
+
+export const name = 'stable-query-client'
+
+export const rule = createRule({
+  name,
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Makes sure that QueryClient is stable',
+      recommended: 'error',
+    },
+    messages: {
+      unstable: [
+        'QueryClient is not stable. It should be either extracted from the component or wrapped in React.useState.',
+        'See https://tkdodo.eu/blog/react-query-fa-qs#2-the-queryclient-is-not-stable',
+      ].join('\n'),
+      fixTo: 'Fix to {{result}}',
+    },
+    hasSuggestions: true,
+    fixable: 'code',
+    schema: [],
+  },
+  defaultOptions: [],
+
+  create(context, _, helpers) {
+    return {
+      NewExpression(node) {
+        if (
+          node.callee.type !== AST_NODE_TYPES.Identifier ||
+          node.callee.name !== 'QueryClient' ||
+          !helpers.isSpecificTanstackQueryImport(
+            node.callee,
+            '@tanstack/react-query',
+          )
```

Comment:
I was uncertain if this rule is relevant to non-react packages, so currently it works only if QueryClient is imported from `@tanstack/react-query`.

Reply:
good call, I think e.g. for solid it doesn't matter.